source/manual/errors.html.md

```diff
@@ -1,109 +1,50 @@
-
-### Intermittent errors without user impact
-
-Frontend applications often see timeouts when talking to the content-store.
-
-There's no or little user impact because the request will be answered by the caching layer.
-
-Example: <https://sentry.io/govuk/app-finder-frontend/issues/352985400>
+## How we categorise errors
 
-Desired behaviour: error is not sent to Sentry. Instead, we rely on Smokey and Icinga checks to make sure we the site functions.
-
-### Intermittent errors with user impact
-
-Publishing applications sometimes see timeouts when talking to publishing-api. This results in the publisher seeing an error page and possibly losing data.
-
-Example: <https://sentry.io/govuk/app-content-tagger/issues/367277928>
-
-Desired behaviour: apps handle these errors better, for example by offloading the work to a Sidekiq worker. Since these errors aren't actionable, they should not be reported to Sentry. They should be tracked in Graphite.
-
-### Intermittent retryable errors
-
-Sidekiq worker sends something to the publishing-api, which times out. Sidekiq retries, the next time it works.
-
-Desired behaviour: errors are not reported to Sentry until retries are exhausted. See [this PR for an example](https://github.com/alphagov/content-data-api/pull/353).
-
-Relevant: https://github.com/getsentry/raven-ruby/pull/784
-
-### Expected environment-based errors
-
-MySQL errors on staging while data sync happens.
-
-Example: <https://sentry.io/govuk/app-whitehall/issues/343619055>
-
-Desired behaviour: our environment is set up such that these errors do not occur.
-
-### Bad request errors
-
-User makes a request the application can't handle ([example][bad-request]).
+Sometimes applications will encounter exceptions. This policy describes what we should do for different types of errors. It was first proposed in [RFC 87](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-087-dealing-with-errors.md).
 
-Often happens in [security checks](https://sentry.io/govuk/app-frontend/issues/400074979).
+### High priority errors
 
-Example: <https://sentry.io/govuk/app-frontend/issues/400074979>
+1. **When something goes wrong, we should be notified**. Applications should report exceptions to Sentry. Applications must not swallow errors.
 
-Desired behaviour: user gets feedback, error is not reported to Sentry
+1. **Notifications should be actionable**. Sentry notifications should be something that requires a developer of the app to do something about it. Not just a piece of information.
 
-[bad-request]: https://sentry.io/govuk/app-service-manual-frontend/issues/400074003
+1. **Applications should not continue to have these errors**. The goal of GOV.UK is that applications should not error. When something goes wrong it should be fixed.
 
-### Incorrect bubbling up of errors
+Errors that should be reported:
```

Comment:
This suggests to me that there are high-priority errors that shouldn't be reported? Which goes a bit against "Applications should report exceptions to Sentry"? 

Is this just a list of what we consider to be a high priority error?

Reply:
Oops, you're right, it's a strange implication left over from a previous draft. Updated 👍.